src/nonlinear.jl

```diff
@@ -27,8 +27,8 @@ function StateSpaceProblem(
     h::htype,
     u0::utype,
     tspan::ttype,
-    params=nothing;
-    obs_noise = StandardGaussian(size(h(u0, params, 0))), # TODO: Might be suboptimal way to get size of obs
+    params = nothing;
+    obs_noise = MvNormal(ones(size(h(u0, params, 0)))), # Assume the default measurement error is MvNormal with identity covariance
```

Comment:
This constructor is deprecated. Covariance matrices should always be matrices. Also you might want to set the element type depending on the output of `h`.

Reply:
Thanks for reminding me of the matrices.
For your later comment, are you suggesting using the differentiable distributions? If so, is there an example that I can have a look at?

Reply:
Hmm, I'm not sure what you refer to here. I only wanted to suggest using something like
```suggestion
    obs_noise = (h0 = h(u0, params, t0); MvNormal(Zeros{eltype(h0)}(length(h0)), I))
```
that uses an efficient covariance matrix (for this type optimized dispatches exist in Distributions) and a non-allocating zero mean vector that respects the element type of the output of `h`.

Reply:
I see. Year, this is for sure a good idea.